src/app/add-customer-dialog/add-customer-dialog.component.ts

```diff
@@ -0,0 +1,78 @@
+import { HttpClient } from '@angular/common/http';
+import { Component, OnInit } from '@angular/core';
+import { Form, FormBuilder, FormGroup } from '@angular/forms';
+import { MatDialogRef } from '@angular/material/dialog';
+import { environment } from 'src/environments/environment';
+import { CustomerDetailComponent } from '../customer-detail/customer-detail.component';
+import { Customer } from '../customers/customer';
+
+@Component({
+  selector: 'app-add-customer-dialog',
+  templateUrl: './add-customer-dialog.component.html',
+  styleUrls: ['./add-customer-dialog.component.css']
+})
+export class AddCustomerDialogComponent implements OnInit {
+  
+  form!: FormGroup;
+  customers!:Customer[];
+    
+  constructor(private formB: FormBuilder,
+    private httpClient:HttpClient,
+    private dialRef: MatDialogRef<CustomerDetailComponent> ){
+      this.form=this.formB.group({
+        RoleID:[],
+        FirstName:[],
+        LastName:[],
+        BirthDate:[],
+        AddressID:[],
+        Email:[],
+        PhoneNumber:[],
+        Password:[]
+      })
+    }
+
+  ngOnInit(): void {
+    this.getCustomers();
+  }  
+
+  getCustomers()
+  {
+    this.httpClient.get<any>(environment.baseUrl+'customers').subscribe(
+    response=>{
+      console.log(response);
+      this.customers=response;
+    }
+    );5
+  }
+
+  cancel() {
+    this.dialRef.close();
+  }
+  
+  Submit()
+  {
+    let user: Customer = {
+      RoleID: Number(this.form.get('RoleID')),
+      FirstName: this.form.get('FirstName')?.value,
+      LastName: this.form.get('LastName')?.value,
+      BirthDate: this.form.get('BirthDate')?.value,
+      AddressID: Number(this.form.get('AddressID')?.value),
+      Email: this.form.get('Email')?.value,
+      PhoneNumber: this.form.get('PhoneNumber')?.value,
+      Password: this.form.get('Password')?.value,
+      CustomerID: 0,
+      GasType: 0,
+      Electricitytype: 0,
+      UserID: 0,
```

Comment:
Use two-way binding for this instead. Then simply submit that customer obj. Apply this to other places as well. Will simplify a lot of code.